e2e/docker/supernode.Dockerfile

```
@@ -0,0 +1,11 @@
+FROM flwr/supernode:nightly
+
+WORKDIR /app
+RUN python -m pip install -U --no-cache-dir\
+    "flwr-datasets["vision"]>=0.1.0,<1.0.0" \
+    torch==2.2.1 \
+    torchvision==0.17.1 \
+    && pyenv rehash
```

Wouldn't it be better to have a `pyproject.toml` and then just copy that? I usually like that pattern more then tracking dependencies in a Dockerfile. Not sure though. I might be missing something.


```suggestion
COPY pyproject.toml ./
RUN python -m pip install -U --no-cache-dir . && pyenv rehash
```

---

@tanertopal That's a valid point. I don't have a strong argument for using this apart from possibly running `pytest` against different versions of `torch` and `torchvision`. Will add `pyproject.toml` to it.